salt/utils/find.py

```diff
@@ -655,10 +659,22 @@ def find(self, path):
                             if result is not None:
                                 yield result
 
-            if depth == self.maxdepth:
+            if self.maxdepth is not None and depth > self.maxdepth:
                 dirs[:] = []
 
 
+def path_depth(path):
+    depth = 0
+    head = path
+    while True:
```

Comment:
There is no recursion; only iteration.

The break condition variable, `tail`, is assigned on every iteration, on line 670.

`head, tail = os.path.split(head)`

`head` is iteratively deconstructed, with the next `tail` removed each time. So `head` iterates to `/` or `.`, while `tail` iterates to `.` or ``.

I did find a bug, though, in the case where path ends in `/`. I'll fix that.